Vue part

Niedzwiedz opened the pull request:
<!-- Reviewable:start -->

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/niedzwiedz/montekombat/6)

<!-- Reviewable:end -->

Niedzwiedz committed:
Work in progress - edit tournament, representers

Edit tournament feature, API done with use of representers.

Niedzwiedz committed:
[Work in progress] Edit tournament and representer

Niedzwiedz committed:
Fix too long line in tournament spec

Niedzwiedz committed:
Some authorization for teams and tournaments

Niedzwiedz committed:
Fix tournament create, joining to tounament

Niedzwiedz committed:
[Work in progress] Generate rounds

Integrate sidekiq, and generate first round

Niedzwiedz committed:
Vue as client side app

Connect to api and display matches

Niedzwiedz committed:
Token based login and logout

Also header problem

Niedzwiedz committed:
Additional files for login and logout

Niedzwiedz committed:
Not completly fixed header problem

shentao commented:
Most of the stuff looks fine. Just please take care of the leftovers – Hello.vue file, old rails views and assets (they are unnecessary, as rails now only works as an API).
In the next commit I would like to see a working version of setting up the default auth header.

Also, please either use the linter or remove it (which is something I don’t recommend).

---

Reviewed 39 of 76 files at r1.
Review status: 39 of 76 files reviewed at latest revision, 13 unresolved discussions.

---

_[Gemfile, line 20 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQr0Z3NNNjnIyIIVXM:-KUQr0Z3NNNjnIyIIVXN:b-816h6l) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/Gemfile#L20)):_

> ``` Ruby
> 
> # Use jquery as the JavaScript library
> gem "jquery-rails"
> ```

Same.

---

_[Gemfile, line 40 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQr-TqYg-c9xUYLX0q:-KUQr-TqYg-c9xUYLX0r:b-knvnre) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/Gemfile#L40)):_

> ``` Ruby
> gem "slim-rails", "~> 3.1.1"
> 
> gem "vuejs-rails"
> ```

Can be removed.

---

_[app/assets/javascripts/application.js, line 24 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQmdDt4FmZ6Z2V1G9P:-KUQmdDuBIqVP_ZGtBq1:b-nbpjc7) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/app/assets/javascripts/application.js#L24)):_

> ``` JavaScript
> 
> 
> var tournaments = new Vue({
> ```

I’m not sure if this file is valid.

---

_[app/views/tournaments/edit.html.slim, line 3 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQmxx9GOo5IRvLgZQv:-KUQmxx9GOo5IRvLgZQw:b-1ec2im) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/app/views/tournaments/edit.html.slim#L3)):_

> ``` Slim
> = render 'shared/error_messages', object: @tournament
> 
> #tournament_edit_form
> ```

And this.

---

_[app/views/tournaments/new.html.slim, line 3 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQmuDww95ZoGeSngCq:-KUQmuDww95ZoGeSngCr:b-jy3gwr) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/app/views/tournaments/new.html.slim#L3)):_

> ``` Slim
> = render 'shared/error_messages', object: @tournament
> 
> #tournament_form
> ```

Same with this one.

---

_[client/src/App.vue, line 17 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQnb-yjDl0BtdBveJt:-KUQnb-yjDl0BtdBveJu:b-lamaqw) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/client/src/App.vue#L17)):_

> ``` Vue
>   components: {
>     Navigation,
>     Hello
> ```

Remove if not used.

---

_[client/src/auth/auth.js, line 14 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQo35qdeWNqlXg8XPe:-KUQo35qdeWNqlXg8XPf:b4zdq5h) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/client/src/auth/auth.js#L14)):_

> ``` JavaScript
> 
>   login (email, password) {
>     axios.post(LOGIN_PATH, {auth: {email: email, password: password}}).then(response => {
> ```

You have the linter config set, but you’re not following its rules. If someone other would run the project, who actually uses a linter, he would get tons of errors. 

---

_[client/src/auth/auth.js, line 26 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQoR25_4m1zpa4BoXX:-KUQoR25_4m1zpa4BoXY:bpfdvmi) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/client/src/auth/auth.js#L26)):_

> ``` JavaScript
>   checkAuth () {
>     var jwt = localStorage.getItem('id_token')
>     if (jwt) { this.user.authenticated = true } else { this.user.authenticated = false }
> ```

For assignments and returns it is actually quite nice to use ternaries.

```
this.user.authenticated = jwt ? true : false
```

But in this example, since you need a boolean, you can do

```
this.user.authenticated = !!jwt
```

---

_[client/src/components/Hello.vue, line 1 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQpdT2eqynz-zekxd9:-KUQpdT2eqynz-zekxdA:bru2pip) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/client/src/components/Hello.vue#L1)):_

> ``` Vue
> <template>
> ```

Remove this file as it is not related to the project.

---

_[client/src/components/Login.vue, line 10 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQpmDNREkzNd9Hnqun:-KUQpmDNREkzNd9Hnquo:b-7shdjh) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/client/src/components/Login.vue#L10)):_

> ``` Vue
>       <input type="password" class="form-control" placeholder="Enter your password" v-model="password">
>     </div>
>     <button @click="submit()"> Login </button>
> ```

Buttons should always contain `type="button"`, otherwise they are treated as submit buttons. This often leads to unpredicted behavior, like refreshing the page on IE.

---

_[client/src/components/Match.vue, line 14 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQq2-c0ntvP41dtri7:-KUQq2-dhpxFQfapSzp3:ba3dfmt) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/client/src/components/Match.vue#L14)):_

> ``` Vue
>       }
>     },
>     methods: {}
> ```

Remove.

---

_[client/src/components/Nav.vue, line 18 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQqIRzHUhMfr5HI4Rl:-KUQqIRzHUhMfr5HI4Rm:b-b4ajk7) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/client/src/components/Nav.vue#L18)):_

> ``` Vue
>     }
>   },
>   computed: {},
> ```

Remove unnecessary keys.

---

_[client/test/unit/specs/Hello.spec.js, line 1 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQqkpyJ0rzst1eOQM3:-KUQqkpyJ0rzst1eOQM4:b-3iyq7t) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/client/test/unit/specs/Hello.spec.js#L1)):_

> ``` JavaScript
> import Vue from 'vue'
> ```

Remove this file.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-:-KUQr5Zj-3iU99RV_Ejk:bvt285)_

<!-- Sent from Reviewable.io -->

Niedzwiedz committed:
Default header

first set of changes for review

Niedzwiedz committed:
Deleted view folder from rails app

Niedzwiedz commented:
Review status: 24 of 94 files reviewed at latest revision, 13 unresolved discussions.

---

_[app/assets/javascripts/application.js, line 24 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQmdDt4FmZ6Z2V1G9P:-KURW-q9bMpWte7ps1ZY:bat62pk) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/app/assets/javascripts/application.js#L24)):_
<details><summary>

<i>Previously, shentao (Damian Dulisz) wrote…</i></summary>



> I’m not sure if this file is valid.
> </details>
> I was playing with vue here. And now I know that it wasn't really vue. Whole application.js is going to be cleared.
> Also all not vue views will be deleted from rails directory.

---

_[client/src/auth/auth.js, line 14 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQo35qdeWNqlXg8XPe:-KURXeZonbdoulXfFZli:b936nwc) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/client/src/auth/auth.js#L14)):_
<details><summary>

<i>Previously, shentao (Damian Dulisz) wrote…</i></summary>



> You have the linter config set, but you’re not following its rules. If someone other would run the project, who actually uses a linter, he would get tons of errors. 
> </details>
> hmm.. I even installed Atom :hurtrealbad: with lint and eslint to see what you're talking about, and nothing in here is shouting at me.  Maybe I configured something wrong, when you will be in the office next time please show me how it should look like 

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6)_

<!-- Sent from Reviewable.io -->

shentao commented:
Reviewed 10 of 36 files at r2.
Review status: 32 of 94 files reviewed at latest revision, 12 unresolved discussions.

---

_[client/src/api.js, line 7 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUS2Tcswk9-7koBmMZ3:-KUS2Tcswk9-7koBmMZ4:bihwjqe) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e817b79a20332ccc46c60a721d230c5e1bd70c7f/client/src/api.js#L7)):_

> ``` JavaScript
> export default {
>   createHeader () {
>     axios.defaults.headers.common['Authorization'] = Auth.getAuthHeader()['Authorization']
> ```

Why the `['Authorization']` at the end? Why not let the method return just the `'Bearer <token_here>`

---

_[client/src/auth/auth.js, line 14 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQo35qdeWNqlXg8XPe:-KUS2mdqryaXlLtbY33O:bz9498t) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/client/src/auth/auth.js#L14)):_
<details><summary>

<i>Previously, Niedzwiedz (Michał Nowicki) wrote…</i></summary>



> hmm.. I even installed Atom :hurtrealbad: with lint and eslint to see what you're talking about, and nothing in here is shouting at me.  Maybe I configured something wrong, when you will be in the office next time please show me how it should look like 
> </details>
> Wasn’t sure if the standard style linter has anything against spaces around curly brackets. False alarm. Still, using the linter is a good idea – it will help you catch some potential bugs and dead code.

---

_[client/src/auth/auth.js, line 3 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUS37JV5D-WJuFiEa3B:-KUS37JW-kcbFDQxvLLx:b8z9xk1) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e817b79a20332ccc46c60a721d230c5e1bd70c7f/client/src/auth/auth.js#L3)):_

> ``` JavaScript
> // import {router} from './../main.js'
> import axios from 'axios'
> import myApi from '../api'
> ```

Why call it myApi and not just api?

---

_[client/src/auth/auth.js, line 15 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUS3T7jVYI53Sr6o9RB:-KUS3T7jVYI53Sr6o9RC:b-9a60tj) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e817b79a20332ccc46c60a721d230c5e1bd70c7f/client/src/auth/auth.js#L15)):_

> ``` JavaScript
> 
>   login (email, password) {
>     axios.post(LOGIN_PATH, {auth: {email: email, password: password}}).then(response => {
> ```

If you have a API abstraction layer created, why are you using axios here directly?

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6)_

<!-- Sent from Reviewable.io -->

Niedzwiedz committed:
Fix for code review

First working usage of interceptors, but it still needs more work

Niedzwiedz commented:
Review status: 32 of 94 files reviewed at latest revision, 11 unresolved discussions.

---

_[Gemfile, line 20 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQr0Z3NNNjnIyIIVXM:-KUSPWINJIiDwuN9gG5V:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/Gemfile#L20)):_
<details><summary>

<i>Previously, shentao (Damian Dulisz) wrote…</i></summary>



> Same.
> </details>
> Done.

---

_[Gemfile, line 40 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQr-TqYg-c9xUYLX0q:-KUSPWlJ8ZzQS9AiQFnh:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/Gemfile#L40)):_
<details><summary>

<i>Previously, shentao (Damian Dulisz) wrote…</i></summary>



> Can be removed.
> </details>
> Done.

---

_[app/assets/javascripts/application.js, line 24 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQmdDt4FmZ6Z2V1G9P:-KUSQA4IvVqJxwNuDTeV:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/app/assets/javascripts/application.js#L24)):_
<details><summary>

<i>Previously, Niedzwiedz (Michał Nowicki) wrote…</i></summary>



> I was playing with vue here. And now I know that it wasn't really vue. Whole application.js is going to be cleared.
> Also all not vue views will be deleted from rails directory.
> </details>
> Done.

---

_[app/views/tournaments/edit.html.slim, line 3 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQmxx9GOo5IRvLgZQv:-KUSRPUfFRn2AEdNrUNB:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/app/views/tournaments/edit.html.slim#L3)):_
<details><summary>

<i>Previously, shentao (Damian Dulisz) wrote…</i></summary>



> And this.
> </details>
> Done.

---

_[app/views/tournaments/new.html.slim, line 3 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQmuDww95ZoGeSngCq:-KUSRtu7CX1wDn-iOF4R:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/app/views/tournaments/new.html.slim#L3)):_
<details><summary>

<i>Previously, shentao (Damian Dulisz) wrote…</i></summary>



> Same with this one.
> </details>
> Done.

---

_[client/src/api.js, line 7 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUS2Tcswk9-7koBmMZ3:-KUSRLpBoz_pMHtcJAZB:bru2nck) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e817b79a20332ccc46c60a721d230c5e1bd70c7f/client/src/api.js#L7)):_
<details><summary>

<i>Previously, shentao (Damian Dulisz) wrote…</i></summary>



> Why the `['Authorization']` at the end? Why not let the method return just the `'Bearer <token_here>`
> </details>
> O.o tru. 

---

_[client/src/App.vue, line 17 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQnb-yjDl0BtdBveJt:-KUSRR3yrRpYGeY-vY2p:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/client/src/App.vue#L17)):_
<details><summary>

<i>Previously, shentao (Damian Dulisz) wrote…</i></summary>



> Remove if not used.
> </details>
> Done.

---

_[client/src/auth/auth.js, line 14 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQo35qdeWNqlXg8XPe:-KUSRrWyFuOYzqJRGwgg:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/client/src/auth/auth.js#L14)):_
<details><summary>

<i>Previously, shentao (Damian Dulisz) wrote…</i></summary>



> Wasn’t sure if the standard style linter has anything against spaces around curly brackets. False alarm. Still, using the linter is a good idea – it will help you catch some potential bugs and dead code.
> </details>
> Done.

---

_[client/src/auth/auth.js, line 3 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUS37JV5D-WJuFiEa3B:-KUSRWtIlIt5wcmNBTGI:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e817b79a20332ccc46c60a721d230c5e1bd70c7f/client/src/auth/auth.js#L3)):_
<details><summary>

<i>Previously, shentao (Damian Dulisz) wrote…</i></summary>



> Why call it myApi and not just api?
> </details>
> Done.

---

_[client/src/auth/auth.js, line 15 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUS3T7jVYI53Sr6o9RB:-KUSiWk5sDqvCvR9g65J:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e817b79a20332ccc46c60a721d230c5e1bd70c7f/client/src/auth/auth.js#L15)):_
<details><summary>

<i>Previously, shentao (Damian Dulisz) wrote…</i></summary>



> If you have a API abstraction layer created, why are you using axios here directly?
> </details>
> Done.

---

_[client/src/components/Hello.vue, line 1 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQpdT2eqynz-zekxd9:-KUSRsPQaZquzxdXUwyh:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/client/src/components/Hello.vue#L1)):_
<details><summary>

<i>Previously, shentao (Damian Dulisz) wrote…</i></summary>



> Remove this file as it is not related to the project.
> </details>
> Done.

---

_[client/test/unit/specs/Hello.spec.js, line 1 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KUQqkpyJ0rzst1eOQM3:-KUSRspRsXzrZYf-1hdD:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/6212e718be0df1925cae996e90792522346ca27c/client/test/unit/specs/Hello.spec.js#L1)):_
<details><summary>

<i>Previously, shentao (Damian Dulisz) wrote…</i></summary>



> Remove this file.
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6)_

<!-- Sent from Reviewable.io -->

Niedzwiedz committed:
Deleted commented out SIGN_UP path from api

Niedzwiedz committed:
Add error status code handling

also functioning beforeEach in routes

Niedzwiedz committed:
Add more views

views for tournaments, teams in tournament and rounds with matches in tournament

Niedzwiedz committed:
Tournament creation form

Need to change data of parent object to show added tournament in list
without reloading tournaments view. Probably vuex will be helpful.

Niedzwiedz committed:
Tournament handling in vuex

Add vuex functionality and handle tournament list, show, post/add and get
from api.

Niedzwiedz committed:
Add games and users to vuex storage

Add games list and users list to vuex storage

Niedzwiedz committed:
Edit tournament in frontend using vuex

Edit basic tournament info, also edit teams and users that are taking
part in edited tournament.

Niedzwiedz committed:
Edit match points

Niedzwiedz committed:
Winner select after match ended

Niedzwiedz committed:
Time issues

Niedzwiedz committed:
Fix date issues, state integration for round

Need to add datepicker with timezone selection

Niedzwiedz committed:
Experiment with winner table

Needs to be refined

kamnog commented:
Reviewed 1 of 76 files at r1, 3 of 36 files at r2, 13 of 18 files at r3.
Review status: 41 of 105 files reviewed at latest revision, 19 unresolved discussions.

---

_[client/src/main.js, line 69 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KVzSUQ91qMeK7ncuuat:-KVzSUQ91qMeK7ncuuau:b-31idx0) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c4be41537e8e25eb796eaa6f0cee5e7e7b7d80fd/client/src/main.js#L69)):_

> ``` JavaScript
> 
> router.beforeEach((to, from, next) => {
>   if (!(to.path === '/login')) {
> ```
1. Check auth
2. If user is authenticated -> `next`
3. If user isn't authenticated -> `login`
4. Edge case -> if is authenticated and wants to jump to `login` -> stay in current site

Also

```
if (condition) {
  if (condition2) {}
  else {}
} else {}
```

is same as `if (condition || condition2)...`

---

_[client/src/components/Error.vue, line 4 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KVzU2KNgNlKTQXNpIbf:-KVzU2KNgNlKTQXNpIbg:bw18y09) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c4be41537e8e25eb796eaa6f0cee5e7e7b7d80fd/client/src/components/Error.vue#L4)):_

> ``` Vue
>   <div>
>     <h1> Error </h1>
>     <p> Sorry, there was an error. </p>
> ```

:dagger: 
Do you use this component anywhere?

---

_[client/src/components/Nav.vue, line 3 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KVzUfAlhpDiTP1nsYwx:-KVzUfAlhpDiTP1nsYwy:bx3akv2) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c4be41537e8e25eb796eaa6f0cee5e7e7b7d80fd/client/src/components/Nav.vue#L3)):_

> ``` Vue
> <template>
> <div>
>   <router-link v-if="user.authenticated" to="/tournaments">Tournaments</router-link>
> ```

https://router.vuejs.org/en/api/router-link.html

`... :to="tournaments"`

same for rest, take a look and tell me if it works :)

---

_[client/src/components/ShowTournament.vue, line 54 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KVzVI_0C_NtfiqUiVrK:-KVzVI_1S_I7ihkA-ixI:byfjuzl) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c4be41537e8e25eb796eaa6f0cee5e7e7b7d80fd/client/src/components/ShowTournament.vue#L54)):_

> ``` Vue
> 
> <style>
>   .tournament-date {
> ```

you've got some CSS here :see_no_evil: :hear_no_evil: :speak_no_evil: 

---

_[client/src/components/TournamentMatch.vue, line 10 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KVzVgQ8F-Np2mevO6gJ:-KVzVgQ8F-Np2mevO6gK:b60kwsb) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c4be41537e8e25eb796eaa6f0cee5e7e7b7d80fd/client/src/components/TournamentMatch.vue#L10)):_

> ``` Vue
>   export default {
>     props: {
>       match: {}
> ```

without passing `match` object, this component doesn't makes sense, right? :P add `required: true` option to this property

http://vuejs.org/v2/guide/components.html#Prop-Validation

---

_[client/src/components/TournamentRound.vue, line 15 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KVzWRfDg23KOYEex7AH:-KVzWRfDg23KOYEex7AI:bv58gli) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c4be41537e8e25eb796eaa6f0cee5e7e7b7d80fd/client/src/components/TournamentRound.vue#L15)):_

> ``` Vue
>     },
>     props: {
>       round: {}
> ```

`required`

---

_[client/src/components/TournamentRounds.vue, line 14 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KVzWZeU_sSBZ7i0C6HS:-KVzWZeU_sSBZ7i0C6HT:bty1jxj) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c4be41537e8e25eb796eaa6f0cee5e7e7b7d80fd/client/src/components/TournamentRounds.vue#L14)):_

> ``` Vue
>     },
>     props: {
>       rounds: {}
> ```

what if `rounds.length === 0`?

---

_[client/src/components/TournamentTeam.vue, line 10 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KVzXfE6_clO-rnhIJZA:-KVzXfE6_clO-rnhIJZB:bv58gli) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c4be41537e8e25eb796eaa6f0cee5e7e7b7d80fd/client/src/components/TournamentTeam.vue#L10)):_

> ``` Vue
>   export default {
>     props: {
>       team: Object
> ```

`required`

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6)_

<!-- Sent from Reviewable.io -->

Niedzwiedz committed:
Sign up and friendly matches index and create form

Sign up feature, list all friendly matches. Create friendly match form
without action to actualy create friendly match.

Niedzwiedz committed:
Fix for CR

Better authentication, add requirements to props, change router links to
point to routes by name and not by path

Niedzwiedz commented:
Review status: 26 of 114 files reviewed at latest revision, 19 unresolved discussions.

---

_[client/src/main.js, line 69 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KVzSUQ91qMeK7ncuuat:-KW239PkVbKRLisk6Awq:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c4be41537e8e25eb796eaa6f0cee5e7e7b7d80fd/client/src/main.js#L69)):_
<details><summary>

<i>Previously, kamnog (Kamil Nogala) wrote…</i></summary>



> 1. Check auth
> 2. If user is authenticated -> `next`
> 3. If user isn't authenticated -> `login`
> 4. Edge case -> if is authenticated and wants to jump to `login` -> stay in current site
> 
> Also
> 
> ```
> if (condition) {
>   if (condition2) {}
>   else {}
> } else {}
> ```
> 
> is same as `if (condition || condition2)...`
> </details>
> Done.

---

_[client/src/components/Error.vue, line 4 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KVzU2KNgNlKTQXNpIbf:-KW1pE6reAgaAPWB2rTx:bb9matm) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c4be41537e8e25eb796eaa6f0cee5e7e7b7d80fd/client/src/components/Error.vue#L4)):_
<details><summary>

<i>Previously, kamnog (Kamil Nogala) wrote…</i></summary>



> :dagger: 
> Do you use this component anywhere?
> </details>
> Yep in api.js I have interceptor that shows this component if 404

---

_[client/src/components/Nav.vue, line 3 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KVzUfAlhpDiTP1nsYwx:-KW27Mp5704OkdRAkp5B:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c4be41537e8e25eb796eaa6f0cee5e7e7b7d80fd/client/src/components/Nav.vue#L3)):_
<details><summary>

<i>Previously, kamnog (Kamil Nogala) wrote…</i></summary>



> https://router.vuejs.org/en/api/router-link.html
> 
> `... :to="tournaments"`
> 
> same for rest, take a look and tell me if it works :)
> 
> </details>
> Done.

---

_[client/src/components/ShowTournament.vue, line 54 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KVzVI_0C_NtfiqUiVrK:-KW27OtNR0J_nyEmEIiG:bb70bo5) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c4be41537e8e25eb796eaa6f0cee5e7e7b7d80fd/client/src/components/ShowTournament.vue#L54)):_
<details><summary>

<i>Previously, kamnog (Kamil Nogala) wrote…</i></summary>



> you've got some CSS here :see_no_evil: :hear_no_evil: :speak_no_evil: 
> </details>
> O.o 

---

_[client/src/components/TournamentMatch.vue, line 10 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KVzVgQ8F-Np2mevO6gJ:-KW295mQI2BiNGQr2QZp:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c4be41537e8e25eb796eaa6f0cee5e7e7b7d80fd/client/src/components/TournamentMatch.vue#L10)):_
<details><summary>

<i>Previously, kamnog (Kamil Nogala) wrote…</i></summary>



> without passing `match` object, this component doesn't makes sense, right? :P add `required: true` option to this property
> 
> http://vuejs.org/v2/guide/components.html#Prop-Validation
> </details>
> Done.

---

_[client/src/components/TournamentRound.vue, line 15 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KVzWRfDg23KOYEex7AH:-KW29G0ss-HyuY4GTyt_:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c4be41537e8e25eb796eaa6f0cee5e7e7b7d80fd/client/src/components/TournamentRound.vue#L15)):_
<details><summary>

<i>Previously, kamnog (Kamil Nogala) wrote…</i></summary>



> `required`
> </details>
> Done.

---

_[client/src/components/TournamentRounds.vue, line 14 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KVzWZeU_sSBZ7i0C6HS:-KW2F_ioRprplFJVvDg7:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c4be41537e8e25eb796eaa6f0cee5e7e7b7d80fd/client/src/components/TournamentRounds.vue#L14)):_
<details><summary>

<i>Previously, kamnog (Kamil Nogala) wrote…</i></summary>



> what if `rounds.length === 0`?
> </details>
> Done.

---

_[client/src/components/TournamentTeam.vue, line 10 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KVzXfE6_clO-rnhIJZA:-KW29fEH9njEOJ7aXyaL:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/c4be41537e8e25eb796eaa6f0cee5e7e7b7d80fd/client/src/components/TournamentTeam.vue#L10)):_
<details><summary>

<i>Previously, kamnog (Kamil Nogala) wrote…</i></summary>



> `required`
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6)_

<!-- Sent from Reviewable.io -->

kamnog commented:
Reviewed 3 of 43 files at r4.
Review status: 29 of 114 files reviewed at latest revision, 20 unresolved discussions.

---

_[client/src/components/MatchInProgress.vue, line 23 at r4](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KW3creI9Gz5ZqACTwbn:-KW3creJmdc6r8jiyIw6:b84jcra) ([raw file](https://github.com/niedzwiedz/montekombat/blob/ed8e87cf9aad810aae2c450cbea60a78cd38a1eb/client/src/components/MatchInProgress.vue#L23)):_

> ``` Vue
>     computed: {
>       tournament () {
>         return this.$store.state.tournaments[this.$store.state.tournaments.findIndex(({ id }) => id === this.$route.params.id)]
> ```

spread this into multiple line please :( its kinda not readable right now :(

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6)_

<!-- Sent from Reviewable.io -->

kamnog commented:
Reviewed 1 of 36 files at r2, 2 of 43 files at r4.
Review status: 31 of 114 files reviewed at latest revision, 20 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6)_

<!-- Sent from Reviewable.io -->

kamnog commented:
Reviewed 1 of 43 files at r4.
Review status: 32 of 114 files reviewed at latest revision, 20 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6)_

<!-- Sent from Reviewable.io -->

shentao commented:
Review status: 32 of 114 files reviewed at latest revision, 20 unresolved discussions.

---

_[client/src/components/MatchInProgress.vue, line 23 at r4](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KW3creI9Gz5ZqACTwbn:-KW3fMZvNYQ3TECLzCBE:bcwzwo4) ([raw file](https://github.com/niedzwiedz/montekombat/blob/ed8e87cf9aad810aae2c450cbea60a78cd38a1eb/client/src/components/MatchInProgress.vue#L23)):_
<details><summary>

<i>Previously, kamnog (Kamil Nogala) wrote…</i></summary>



> spread this into multiple line please :( its kinda not readable right now :(
> </details>
> You could move this into a getter. 

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6)_

<!-- Sent from Reviewable.io -->

Niedzwiedz committed:
Fix too long lines in methods that take data from state.

Fix too long lines in computed part of MatchInProgress, ShowTournament
and TournamentRound components.

Niedzwiedz commented:
Review status: 31 of 114 files reviewed at latest revision, 20 unresolved discussions.

---

_[client/src/components/MatchInProgress.vue, line 23 at r4](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KW3creI9Gz5ZqACTwbn:-KW4B9qGUXjUSPgp2tcl:b-rau55c) ([raw file](https://github.com/niedzwiedz/montekombat/blob/ed8e87cf9aad810aae2c450cbea60a78cd38a1eb/client/src/components/MatchInProgress.vue#L23)):_
<details><summary>

<i>Previously, shentao (Damian Dulisz) wrote…</i></summary>



> You could move this into a getter. 
> </details>
> @shentao Cool idea, but i need to pass argument to this getter, and from what I found in the hurr durr internets - getters in vuex don't take arguments. Maybe I am wrong or maybe should I override this by creating state for current tournament and change it everytime user switches tournaments? But it seems kinda crazy. For now I just spreaded it in multiple lines.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6)_

<!-- Sent from Reviewable.io -->

shentao commented:
Reviewed 1 of 36 files at r2.
Review status: 32 of 114 files reviewed at latest revision, 11 unresolved discussions.

---

_[client/src/components/MatchInProgress.vue, line 23 at r4](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KW3creI9Gz5ZqACTwbn:-KW4OcafSPeueHMCbiHM:b-ucuwqj) ([raw file](https://github.com/niedzwiedz/montekombat/blob/ed8e87cf9aad810aae2c450cbea60a78cd38a1eb/client/src/components/MatchInProgress.vue#L23)):_
<details><summary>

<i>Previously, Niedzwiedz (Michał Nowicki) wrote…</i></summary>



> @shentao Cool idea, but i need to pass argument to this getter, and from what I found in the hurr durr internets - getters in vuex don't take arguments. Maybe I am wrong or maybe should I override this by creating state for current tournament and change it everytime user switches tournaments? But it seems kinda crazy. For now I just spreaded it in multiple lines.
> </details>
> 
> ``` javascript
> tournament: ({ tournaments, route }) => tournaments.find(({ id }) => id === route.params.id)
> ```

Just make sure you use `vuex-router-sync`.

Also never use `var`. Use `const` or `let`. Vars are not block scoped. You can read about this in You Don’t Know JS. 

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6)_

<!-- Sent from Reviewable.io -->

shentao commented:
Review status: 32 of 114 files reviewed at latest revision, 13 unresolved discussions.

---

_[client/src/components/MatchInProgress.vue, line 32 at r5](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KW4QRwPyT1tpmz3KapO:-KW4QRwPyT1tpmz3KapP:bwwuokj) ([raw file](https://github.com/niedzwiedz/montekombat/blob/366c18ae344bbfedaf152eb06d511013834fc6a8/client/src/components/MatchInProgress.vue#L32)):_

> ``` Vue
>           id === this.$route.params.match_id
>         )
>         return this.$store.state.matches[index]
> ```

As above and applies to what’s below.

---

_[client/src/components/MatchInProgress.vue, line 46 at r5](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KW4QhReVXx8FRS4q9u4:-KW4QhReVXx8FRS4q9u5:blo378h) ([raw file](https://github.com/niedzwiedz/montekombat/blob/366c18ae344bbfedaf152eb06d511013834fc6a8/client/src/components/MatchInProgress.vue#L46)):_

> ``` Vue
>         this.$store.dispatch('finishThisMatch', { points1: this.match.points_for_team1, points2: this.match.points_for_team2, matchId: this.$route.params.match_id })
>         if (this.match.points_for_team1 > this.match.points_for_team2) {
>           this.match.winner = this.match.team_1
> ```

Isn’t winner a computed value coming from vuex? You shouldn’t ever directly mutate state objects in components. Use vuex mutations for this.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6)_

<!-- Sent from Reviewable.io -->

shentao commented:
Review status: 32 of 114 files reviewed at latest revision, 14 unresolved discussions.

---

_[client/src/components/MatchInProgress.vue, line 42 at r5](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KW4R0YASmSz-2bz5rKI:-KW4R0YB_UNqNqux2Lk6:b-v3y1o9) ([raw file](https://github.com/niedzwiedz/montekombat/blob/366c18ae344bbfedaf152eb06d511013834fc6a8/client/src/components/MatchInProgress.vue#L42)):_

> ``` Vue
>     },
>     methods: {
>       endMatch () {
> ```

This method is too complex. I believe most of the stuff could be done in the action you dispatch.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6)_

<!-- Sent from Reviewable.io -->

Niedzwiedz committed:
Creation of friendly match with match store problem, backend works

Creation of friendly match, store is uploaded but component doesnt see it

Niedzwiedz committed:
Fully working friendly match creation

Friendly match creation and display on front and backend, add component
for editing tournament

Niedzwiedz committed:
Fix for CR

Add getters for tournament, match, round

Niedzwiedz committed:
Fix for CR

Add vue-router-sync. Getters for tournament, match and round based on
vue-router-sync. Better endMatch method in MatchInProgress.

Niedzwiedz commented:
Review status: 30 of 116 files reviewed at latest revision, 12 unresolved discussions.

---

_[client/src/components/MatchInProgress.vue, line 23 at r4](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KW3creI9Gz5ZqACTwbn:-KWctqJ-Nq7Wr7rDRMiP:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/ed8e87cf9aad810aae2c450cbea60a78cd38a1eb/client/src/components/MatchInProgress.vue#L23)):_
<details><summary>

<i>Previously, shentao (Damian Dulisz) wrote…</i></summary>



> ``` javascript
> tournament: ({ tournaments, route }) => tournaments.find(({ id }) => id === route.params.id)
> ```
> 
> Just make sure you use `vuex-router-sync`.
> 
> Also never use `var`. Use `const` or `let`. Vars are not block scoped. You can read about this in You Don’t Know JS. 
> </details>
> Done.

---

_[client/src/components/MatchInProgress.vue, line 32 at r5](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KW4QRwPyT1tpmz3KapO:-KWctsOEnzlwvU4J5tUm:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/366c18ae344bbfedaf152eb06d511013834fc6a8/client/src/components/MatchInProgress.vue#L32)):_
<details><summary>

<i>Previously, shentao (Damian Dulisz) wrote…</i></summary>



> As above and applies to what’s below.
> </details>
> Done.

---

_[client/src/components/MatchInProgress.vue, line 42 at r5](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KW4R0YASmSz-2bz5rKI:-KWh5-7tfzwy4G4x1r9t:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/366c18ae344bbfedaf152eb06d511013834fc6a8/client/src/components/MatchInProgress.vue#L42)):_
<details><summary>

<i>Previously, shentao (Damian Dulisz) wrote…</i></summary>



> This method is too complex. I believe most of the stuff could be done in the action you dispatch.
> </details>
> Done.

---

_[client/src/components/MatchInProgress.vue, line 46 at r5](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6#-KW4QhReVXx8FRS4q9u4:-KWh53WikNzio0_4tChr:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/366c18ae344bbfedaf152eb06d511013834fc6a8/client/src/components/MatchInProgress.vue#L46)):_
<details><summary>

<i>Previously, shentao (Damian Dulisz) wrote…</i></summary>



> Isn’t winner a computed value coming from vuex? You shouldn’t ever directly mutate state objects in components. Use vuex mutations for this.
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/6)_

<!-- Sent from Reviewable.io -->
